### PR TITLE
Test: Add tests

### PR DESCRIPTION
**Pull request by Ramimashkouk**

### CODE REVIEW

**Review by RLKRo — COMMENTED** [review]
It feels like there are some overlapping / redundant tests.

File: backend/df_designer/app/tests/api/test_flows.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    mocker.patch("app.api.api_v1.endpoints.flows.read_conf", return_value=OmegaConf.create({"foo": "bar"}))
+    response = await flows_get()
+    assert response["status"] == "ok"
+
+
+@pytest.mark.asyncio
+async def test_flows_post(mocker):
+    mocker.patch("app.api.api_v1.endpoints.flows.write_conf", return_value={})
+    response = await flows_post({"foo": "bar"})
+    assert response["status"] == "ok"
```
[the remark this answers is not in the document]
Comment (Ramimashkouk) [reply backend/df_designer/app/tests/api/test_flows.py:19]: I agree about the first point. I'll add an assertion for `data` key.

About second point, well, the `test_api_integration::test_flows` indeed does the same thing as tests in `test_flows.py`. However, it also checks real reading and writing from DB. So, the `integration test` would check the integration between the endpoints and database, while the `unit tests` would ensure that the internal logic are run and the response is valid and that no unintentional mistakes occur while making changes to endpoints. `unit tests` are faster and great to run more often than `integration ones`.

File: backend/df_designer/app/cli.py [hunk 2]
```diff
@@ -56,8 +60,13 @@ def build_bot(build_id: int, project_dir: str = settings.work_directory, preset:
...
 @cli.command("build_scenario")
-def build_scenario(build_id: int, project_dir: str = "."):
-    asyncio.run(translator(build_id=build_id, project_dir=project_dir))
+def build_scenario(build_id: int, project_dir: str = ".", call_from_open_event_loop: bool = False):
+    if call_from_open_event_loop:
+        loop = asyncio.get_event_loop()
+        loop.create_task(translator(build_id=build_id, project_dir=project_dir))
+        loop.run_until_complete(asyncio.wait([], return_when=asyncio.FIRST_COMPLETED))
+    else:
+        asyncio.run(translator(build_id=build_id, project_dir=project_dir))
```
[the remark this answers is not in the document]
Comment (Ramimashkouk) [reply backend/df_designer/app/cli.py:69]: There was a warning which still pops up when testing as [shown here](https://github.com/deeppavlov/dialog_flow_designer/actions/runs/9006617479/job/24744549196?pr=26). 
I thought that several event loops overlapping is the cause of the warning, this's why I added `call_from_open_event_loop`. However, this seemed to solve it for a while but the warning popped up again.

File: backend/df_designer/app/tests/integration/test_api_integration.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    try:
+        await asyncio.wait_for(
+            process_manager.processes[process_manager.last_id].process.wait(), timeout=4
+        )  # TODO: Consider making this timeout configurable
+    except asyncio.exceptions.TimeoutError as exc:
+        if expected_end_status in ["alive", "running"]:
+            logger.debug("Loop process timed out. Expected behavior.")
+        else:
+            logger.debug("Process with expected end status '%s' timed out with status 'running'.", expected_end_status)
+            raise exc
```
[the remark this answers is not in the document]
Comment (Ramimashkouk) [reply backend/df_designer/app/tests/integration/test_api_integration.py:27]: Resolved in 34781574315f6375f55368ca8c9545c52974ffb6

File: backend/df_designer/app/tests/services/test_process.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    # def test_update_db_info(self, run_process):
+    #     process = await run_process("echo 'Hello df_designer'")
+    #     process.update_db_info()
+
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize(
+        "cmd_to_run, status",
+        [
+            ("sleep 10000", "running"),
+            ("cat /non_existing_file", "failed"),
```
[the remark this answers is not in the document]
Comment (Ramimashkouk) [reply backend/df_designer/app/tests/services/test_process.py:20]: Resolved in 34781574315f6375f55368ca8c9545c52974ffb6

**Pull request merged**